applications/app_ppm.c

```diff
@@ -333,6 +338,32 @@ static THD_FUNCTION(ppm_thread, arg) {
 			}
 			break;
 
+		case PPM_CTRL_TYPE_PID_POSITION: // make this an actual type.
+			if (fabsf(servo_val) < 0.001) {
+				pulses_without_power++;
+			}
+
+			float angle = servo_val * 360;
+
+			if (!(pulses_without_power < MIN_PULSES_WITHOUT_POWER && config.safe_start)) {
+				// try to more intelligently safe start by waiting until 
```

Comment:
I think safe start makes very little sense in position control mode. It will be confusing if it does not work until you give just the right input and then it suddenly starts working. It is also likely to cause more harm than good if it is used for e.g. a steering servo. Any glitch or fault will make the steering not work until you give it just the right range of input to start working again. I think it is best to just skip the safe start entirely for position control.

Reply:
So you can just turn safe start off if you don't want it. But most applications I think you want to disable position control if the signal is lost. (battlebots seems the most interested in this for big swing-y or lifter weapons.) You can just disable safe start on fault if that is an issue for you for something like a steering servo.

I want to make sure that you don't get a fast, potentially violent rotation immediately after power up as it tries to go to the right position. 
I want your hands on the controls where you can try to do something/away from the motor. 
Wiggling the sticks after power up when you want things to move is a fairly natural thing to do. And as you pass through the position it will "latch on".

I can make it configurable if you really think that is needed though.

Reply:
As of right now, to start position control, you need to:
1. Return the control stick to the "resting" position (center ppm for 180 mode, minimum ppm for 360 mode) - your remote should rest in this position.
2. Sweep the control until you hit the position of the motor and it starts tracking.

I think this provides the most amount of safety and convenience, as long as you know what you need to do.
It is possible to disable the first "safe start"  part of this, but not the position matching.

Reply:
> I want to make sure that you don't get a fast, potentially violent rotation immediately after power up as it tries to go to the right position.

100% agreed with the need for this.

Reply:
@vedderb 
What are your thoughts on the safe start as it stands now? especially with the proposed vesc tool help section? Do you think I need to add a setting to disable the position based safe start?

Reply:
Also, if you didn't move the motor from when you powered it off the last time and it was centered. The regular safe start and position safe start windows line up and the motor starts position control right away.